crates/rustc_codegen_spirv/src/symbols.rs

```diff
@@ -596,6 +600,147 @@ fn parse_image_type(
     }))
 }
 
+fn parse_capability_list(arg: &NestedMetaItem) -> Result<Vec<Capability>, ParseAttrError> {
+    let list = match arg.meta_item_list() {
+        Some(list) => list,
+        None => return Err((arg.span(), "`#[spirv(capabilities())]` requires a list.".to_string())),
+    };
+
+    Ok(list.iter().filter_map(NestedMetaItem::ident).filter_map(ident_to_capability).collect())
+}
+
+fn ident_to_capability(ident: Ident) -> Option<Capability> {
+    Some(match &*ident.to_string() {
```

Comment:
`to_string` is unnecessarily expensive (only `proc_macro` suffers from this limitation).

But you don't even need to match on `ident.as_str()` - check out what `const BUILTINS` & friends do, I suspect you want something similar.